### PR TITLE
Add DarkBrewery theme

### LINKED ISSUE #3034: Add Darkbrewery editor theme
### Your idea:

Now that the basic CodeMirror theme functionality is in place, we can extend that functionality to add our own themes.

I suggest that we start with the Darkbrewery theme, which has been popular on the subreddit.

DarkBrewery styling is as follows:
```
@-moz-document domain("homebrewery.naturalcrit.com") {
body {
    ;
}

/* Main BG color and normal text color */
 .CodeMirror {
    background: #293134;
    color: #91A6AA;
}

/* Brew BG */
 .brewRenderer {
    background-color: #29313;
}
/* Blinking cursor */
 .CodeMirror-cursor {
    border-left: 1px solid #e0e2e4;
}
/* Bars at the top */
 .snippetBar {
    background-color: #2F393C;
    color: white;
}
nav {
    background-color: #293134;
}
nav .navItem {
    background-color: #293134;
}
.snippetBar .snippetGroup .dropdown {
    background-color: #2F393C;
}
/* Line number stuff */
 .CodeMirror-gutter-elt {
    color: #81969A;
}
.CodeMirror-linenumber {
    background-color: #293134;
}
.CodeMirror-gutter {
    background-color: #293134;
}
/* column splits */
 .editor .codeEditor .columnSplit {
    font-style: italic;
    color: inherit;
    background-color:#1f5763;
    border-bottom: #299 solid 1px;
}
/* Colors for headings and such */
 .cm-s-default .cm-header {
    color: #c51b1b;
    -webkit-text-stroke-width: 0.1px; 
  	-webkit-text-stroke-color: #000; 
    /* ###Headings */
}
.cm-header, .cm-strong {
    font-weight: bold;
    color: #309dd2;
    /* bold points */
}
.cm-s-default .cm-link {
    color: #dd6300;
    /* Link headings */
}
.cm-s-default .cm-string {
    color: #aa8261;
    /* links */
}
.cm-s-default .cm-def {
color:#2986cc;
}           /*@import*/
.cm-s-default .cm-variable-2 {
    color: #3cbf30;
    /* Bullets and such */
}
.editor .codeEditor .block {
    color: #e3e3e3;
    /* blocks */
}
.editor .codeEditor .inline-block {
    color: #e3e3e3;
    /* inline blocks */
}
.cm-s-default .cm-tag {
    color: #e3ff00;
    /* Tags (divs) */
}
.cm-s-default .cm-attribute {
    color: #e3ff00;
}
.cm-s-default .cm-atom {
    color:#000;
}
.cm-s-default .cm-qualifier{
    color:#ee1919;
}
.cm-s-default .cm-comment{
    color:#bbc700;
}
.cm-s-default .cm-keyword {
	color:#c302df;
    background-color:#b1b1b1;
}
.cm-s-default .cm-property.cm-error {
	color:#c50202;
}

.CodeMirror-foldmarker {
	color:#f0ff00;
}

.editor .codeEditor .pageLine {
    /* New page  */
  background: #000;
    color:#000;
    border-bottom: 1px solid #fff;
}

.cm-s-default .cm-builtin {
	color:#fff;
}

}
```

### PR DESCRIPTION
**Pull request by G-Ambatte**
This PR resolves #3034.

This PR modifies the build script to copy the content of a new folder `./themes/codeMirror/customThemes/` to the Editor Themes build folder, which already has the content of the CodeMirror NPM package's internal `theme` folder copied to it. The modified script then builds the content of the `editorThemes.json` file from the build folder, not the `node_modules` folder as previously.

This allows custom CSS to be created and added to the Homebrewery Editor themes, exactly as if it were one of the CodeMirror supplied themes.

### CODE REVIEW

**Comment by G-Ambatte** [comment]
While I have been able to test that the build script correctly creates `editorThemes.json` with the new `darkbrewery-v301` file, I have not yet been able to test if the contents of that file are correctly applied in the editor when that theme is selected. As such, I have marked this PR as Draft/WIP until such time as I can access my normal dev machine.
Assuming everything works as expected, my intention is to then convert this to a full PR and flag for review by @calculuschild and @5e-Cleric.

**Comment by G-Ambatte** [comment]
From my local dev install:

image

Pinging @5e-Cleric for confirmation that this is the expected result of the DarkBrewery theme.

**Comment by 5e-Cleric** [comment]
I cna confirm

**Comment by 5e-Cleric** [comment]
What color do mustache blocks have right now? i think i did not get around to color those but i could be mistaken.

I'm assuming you have fixed all contrast issues, but confirming is always a good idea.

We could also benefit from nesting here. Even native CSS nesting.

**Review by Gazook89 — CHANGES_REQUESTED** [review]
This is a cool setup.  I think it's a logical extension the editor themes PR.  

@5e-Cleric had asked what the styling for the moustache syntax looked like, and I think it's worth reviewing.  Screenshot below. The purple and the green likely needs to be tweaked before pushing it through.  The red is fine for readability, but could also be improved to fit better with the themes defined colors.

image

EDIT:

This is because the base theme has `:not(.cm-comment)` so the specificity is higher.  So the base theme colors for moustache syntax comes through, overwriting the darkbrewery rules for those elements.

image

So the Darkbrewery file should add that same pseudo-class to those rules and it should be good.

File: themes/codeMirror/customThemes/darkbrewery-v301.css [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    color: inherit;
+    background-color:#1f5763;
+    border-bottom: #299 solid 1px;
+}
+/* Colors for headings and such */
+.cm-s-default .cm-header {
+    color: #c51b1b;
+    -webkit-text-stroke-width: 0.1px; 
+    -webkit-text-stroke-color: #000; 
+    /* ###Headings */
```
Comment (Gazook89) [inline themes/codeMirror/customThemes/darkbrewery-v301.css:55]: There are several places in this CSS where the comment is coming *after* the content it describes, though it is still inside the relevant brackets.  This really threw me off.  I think all those comments should come before their relevant context.

**Comment by 5e-Cleric** [comment]
Didn't think this theme as something that would get its way into the actual site, that's why it's imperfect, so yeah, let's remake everything, with nesting, proper color groups, and comments.

If you don't want to have to come up with new colors for everything, simply setting mustache blocks and injections to a lighter variant of themselves should be fine.

**Review by calculuschild — CHANGES_REQUESTED** [review]
This seems to be working pretty well. I would just request to clean up the comments as per @Gazook89 's note above.

After merging this PR, the next step I think should be to separate out the portions that modify the core HB site so they can be applied to every "dark" theme. It feels odd that this one codemirror theme has this secret unique behavior that bleeds over to affect the rest of the site and the others don't.

**Comment by 5e-Cleric** [comment]
Color wise, i propose these two colors for block and inline mustache elements:
blocks: #ccf781
inline:  #90ff90

DL: pink
DD: #99aaff

File: themes/codeMirror/customThemes/darkbrewery-v301.css (hunk 1 above)
Comment (G-Ambatte) [reply themes/codeMirror/customThemes/darkbrewery-v301.css:55]: Moved the comments to be consistently placed.

**Review by calculuschild — APPROVED** [review]

**Comment by calculuschild** [comment]
Looks good! Thanks @G-Ambatte

**Pull request merged**